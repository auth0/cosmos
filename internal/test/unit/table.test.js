import React from 'react'
import { shallow, mount, render } from 'enzyme'
import { Table, Avatar } from '@auth0/cosmos'

const defaultItems = [
  {
    name: 'Harry Kane',
    goals: 6,
    assists: 0,
    country: 'ðŸ‡¬ðŸ‡§',
    image: 'https://twitter-avatar.now.sh/HKane'
  },
  {
    name: 'Romelu Lukaku',
    goals: 4,
    assists: 1,
    country: 'ðŸ‡§ðŸ‡ª',
    image: 'https://twitter-avatar.now.sh/Romelu_lukaku9'
  },
  {
    name: 'Antoine Griezmann',
    goals: 4,
    assists: 2,
    country: 'ðŸ‡«ðŸ‡·',
    image: 'https://twitter-avatar.now.sh/AntoGriezmann'
  },
  {
    name: 'Ivan PeriÅ¡iÄ‡',
    goals: 3,
    assists: 1,
    country: 'ðŸ‡­ðŸ‡·',
    image: 'https://twitter-avatar.now.sh/ivanperisic44'
  }
]

const tableFactory = ({
  items = defaultItems,
  onSort = () => {},
  onRowClick = () => {},
  wrapper = shallow
} = {}) =>
  wrapper(
    <Table items={items} onRowClick={onRowClick} onSort={onSort}>
      <Table.Column field="image" width="50px">
        {item => <Avatar type="user" image={item.image} />}
      </Table.Column>
      <Table.Column field="name" title="Name" width="30%" sortable />
      <Table.Column field="country" title="Country" />
      <Table.Column field="goals" title="Goals" />
      <Table.Column field="assists" title="Assists" />
    </Table>
  )

describe('Table', () => {
  it('renders properly', () => {
    const wrapper = tableFactory()
    expect(wrapper).toMatchSnapshot()
  })

  it('calls onRowClick', () => {
    const tableParams = { onRowClick: jest.fn(), wrapper: mount }
    const wrapper = tableFactory(tableParams)

    wrapper
      .find('tr')
      .last()
      .simulate('click')

    expect(tableParams.onRowClick).toHaveBeenCalledTimes(1)
  })

  it('calls onSort', () => {
    const tableParams = { onSort: jest.fn(), wrapper: mount }
    const wrapper = tableFactory(tableParams)

    const tableHeader = wrapper.find('th').at(1)

    tableHeader.simulate('click')
    expect(tableParams.onSort).toHaveBeenCalledWith('name', 'desc')
  })

  it('provides automatic sorting', () => {
    const data = [1, 3, 5, 7, 8, 0, 2, 4, 6, 9].map(number => ({ number }))

    const wrapper = mount(
      <Table items={data}>
        <Table.Column sortable field="number" title="Number" />
      </Table>
    )
    const tableItems = () => wrapper.find('td').map(x => x.text())

    const ascending = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9']
    expect(tableItems()).toEqual(ascending)

    wrapper
      .find('th')
      .first()
      .simulate('click')

    const descending = ['9', '8', '7', '6', '5', '4', '3', '2', '1', '0']
    expect(tableItems()).toEqual(descending)
  })

  it('sorts with custom compare function', () => {
    const wrapper = mount(
      <Table items={defaultItems}>
        <Table.Column field="goals" title="Goals" sortable />
        <Table.Column field="assists" title="Assists" sortable />
        <Table.Column
          field="points"
          title="Points"
          sortable
          comparator={(a, b) => {
            return a.goals + a.assists - b.goals - b.assists
          }}
        >
          {item => item.goals + item.assists}
        </Table.Column>
      </Table>
    )

    const getItemsAtColumn = column =>
      wrapper
        .find('tr')
        .map(tr => tr.find('td').map(td => td.text())[column])
        .slice(1) // Remove the header (which is a <tr> as well)

    expect(getItemsAtColumn(0)).toEqual(['3', '4', '4', '6'])
    expect(getItemsAtColumn(1)).toEqual(['1', '1', '2', '0'])
    expect(getItemsAtColumn(2)).toEqual(['4', '5', '6', '6'])

    wrapper
      .find('th')
      .last()
      .simulate('click')

    expect(getItemsAtColumn(0)).toEqual(['3', '4', '6', '4'])
    expect(getItemsAtColumn(1)).toEqual(['1', '1', '0', '2'])
    expect(getItemsAtColumn(2)).toEqual(['4', '5', '6', '6'])
  })
})
