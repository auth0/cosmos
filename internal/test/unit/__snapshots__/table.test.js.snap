// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders properly 1`] = `
<styled.div>
  <styled.table
    data-cosmos-key="table"
    emptyMessage="There are no items to display"
    items={
      Array [
        Object {
          "assists": 0,
          "country": "ðŸ‡¬ðŸ‡§",
          "goals": 6,
          "image": "https://twitter-avatar.now.sh/HKane",
          "name": "Harry Kane",
        },
        Object {
          "assists": 1,
          "country": "ðŸ‡§ðŸ‡ª",
          "goals": 4,
          "image": "https://twitter-avatar.now.sh/Romelu_lukaku9",
          "name": "Romelu Lukaku",
        },
        Object {
          "assists": 2,
          "country": "ðŸ‡«ðŸ‡·",
          "goals": 4,
          "image": "https://twitter-avatar.now.sh/AntoGriezmann",
          "name": "Antoine Griezmann",
        },
        Object {
          "assists": 1,
          "country": "ðŸ‡­ðŸ‡·",
          "goals": 3,
          "image": "https://twitter-avatar.now.sh/ivanperisic44",
          "name": "Ivan PeriÅ¡iÄ‡",
        },
      ]
    }
    onRowClick={[Function]}
    onSort={null}
    rows={
      Array [
        <styled.tr
          data-cosmos-key="table.row"
          onClick={[Function]}
        >
          <styled.td
            column={
              Object {
                "children": [Function],
                "field": "image",
                "sortable": false,
                "width": "50px",
              }
            }
          >
            <Avatar
              image="https://twitter-avatar.now.sh/HKane"
              size="medium"
              type="user"
            />
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "name",
                "sortable": true,
                "title": "Name",
                "width": "30%",
              }
            }
          >
            Harry Kane
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "country",
                "sortable": false,
                "title": "Country",
              }
            }
          >
            ðŸ‡¬ðŸ‡§
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "goals",
                "sortable": false,
                "title": "Goals",
              }
            }
          >
            6
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "assists",
                "sortable": false,
                "title": "Assists",
              }
            }
          >
            0
          </styled.td>
        </styled.tr>,
        <styled.tr
          data-cosmos-key="table.row"
          onClick={[Function]}
        >
          <styled.td
            column={
              Object {
                "children": [Function],
                "field": "image",
                "sortable": false,
                "width": "50px",
              }
            }
          >
            <Avatar
              image="https://twitter-avatar.now.sh/Romelu_lukaku9"
              size="medium"
              type="user"
            />
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "name",
                "sortable": true,
                "title": "Name",
                "width": "30%",
              }
            }
          >
            Romelu Lukaku
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "country",
                "sortable": false,
                "title": "Country",
              }
            }
          >
            ðŸ‡§ðŸ‡ª
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "goals",
                "sortable": false,
                "title": "Goals",
              }
            }
          >
            4
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "assists",
                "sortable": false,
                "title": "Assists",
              }
            }
          >
            1
          </styled.td>
        </styled.tr>,
        <styled.tr
          data-cosmos-key="table.row"
          onClick={[Function]}
        >
          <styled.td
            column={
              Object {
                "children": [Function],
                "field": "image",
                "sortable": false,
                "width": "50px",
              }
            }
          >
            <Avatar
              image="https://twitter-avatar.now.sh/AntoGriezmann"
              size="medium"
              type="user"
            />
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "name",
                "sortable": true,
                "title": "Name",
                "width": "30%",
              }
            }
          >
            Antoine Griezmann
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "country",
                "sortable": false,
                "title": "Country",
              }
            }
          >
            ðŸ‡«ðŸ‡·
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "goals",
                "sortable": false,
                "title": "Goals",
              }
            }
          >
            4
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "assists",
                "sortable": false,
                "title": "Assists",
              }
            }
          >
            2
          </styled.td>
        </styled.tr>,
        <styled.tr
          data-cosmos-key="table.row"
          onClick={[Function]}
        >
          <styled.td
            column={
              Object {
                "children": [Function],
                "field": "image",
                "sortable": false,
                "width": "50px",
              }
            }
          >
            <Avatar
              image="https://twitter-avatar.now.sh/ivanperisic44"
              size="medium"
              type="user"
            />
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "name",
                "sortable": true,
                "title": "Name",
                "width": "30%",
              }
            }
          >
            Ivan PeriÅ¡iÄ‡
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "country",
                "sortable": false,
                "title": "Country",
              }
            }
          >
            ðŸ‡­ðŸ‡·
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "goals",
                "sortable": false,
                "title": "Goals",
              }
            }
          >
            3
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "assists",
                "sortable": false,
                "title": "Assists",
              }
            }
          >
            1
          </styled.td>
        </styled.tr>,
      ]
    }
    sortDirection="asc"
  >
    <TableHeader
      columns={
        Array [
          Object {
            "children": [Function],
            "field": "image",
            "sortable": false,
            "width": "50px",
          },
          Object {
            "field": "name",
            "sortable": true,
            "title": "Name",
            "width": "30%",
          },
          Object {
            "field": "country",
            "sortable": false,
            "title": "Country",
          },
          Object {
            "field": "goals",
            "sortable": false,
            "title": "Goals",
          },
          Object {
            "field": "assists",
            "sortable": false,
            "title": "Assists",
          },
        ]
      }
      onSort={[Function]}
      sortDirection="asc"
      sortingColumn={
        Object {
          "field": "name",
          "sortable": true,
          "title": "Name",
          "width": "30%",
        }
      }
    />
    <styled.tbody
      data-cosmos-key="table.body"
    >
      <styled.tr
        data-cosmos-key="table.row"
        key="row-0"
        onClick={[Function]}
      >
        <styled.td
          column={
            Object {
              "children": [Function],
              "field": "image",
              "sortable": false,
              "width": "50px",
            }
          }
          key="image"
        >
          <Avatar
            image="https://twitter-avatar.now.sh/HKane"
            size="medium"
            type="user"
          />
        </styled.td>
        <styled.td
          column={
            Object {
              "field": "name",
              "sortable": true,
              "title": "Name",
              "width": "30%",
            }
          }
          key="name"
        >
          Harry Kane
        </styled.td>
        <styled.td
          column={
            Object {
              "field": "country",
              "sortable": false,
              "title": "Country",
            }
          }
          key="country"
        >
          ðŸ‡¬ðŸ‡§
        </styled.td>
        <styled.td
          column={
            Object {
              "field": "goals",
              "sortable": false,
              "title": "Goals",
            }
          }
          key="goals"
        >
          6
        </styled.td>
        <styled.td
          column={
            Object {
              "field": "assists",
              "sortable": false,
              "title": "Assists",
            }
          }
          key="assists"
        >
          0
        </styled.td>
      </styled.tr>
      <styled.tr
        data-cosmos-key="table.row"
        key="row-1"
        onClick={[Function]}
      >
        <styled.td
          column={
            Object {
              "children": [Function],
              "field": "image",
              "sortable": false,
              "width": "50px",
            }
          }
          key="image"
        >
          <Avatar
            image="https://twitter-avatar.now.sh/Romelu_lukaku9"
            size="medium"
            type="user"
          />
        </styled.td>
        <styled.td
          column={
            Object {
              "field": "name",
              "sortable": true,
              "title": "Name",
              "width": "30%",
            }
          }
          key="name"
        >
          Romelu Lukaku
        </styled.td>
        <styled.td
          column={
            Object {
              "field": "country",
              "sortable": false,
              "title": "Country",
            }
          }
          key="country"
        >
          ðŸ‡§ðŸ‡ª
        </styled.td>
        <styled.td
          column={
            Object {
              "field": "goals",
              "sortable": false,
              "title": "Goals",
            }
          }
          key="goals"
        >
          4
        </styled.td>
        <styled.td
          column={
            Object {
              "field": "assists",
              "sortable": false,
              "title": "Assists",
            }
          }
          key="assists"
        >
          1
        </styled.td>
      </styled.tr>
      <styled.tr
        data-cosmos-key="table.row"
        key="row-2"
        onClick={[Function]}
      >
        <styled.td
          column={
            Object {
              "children": [Function],
              "field": "image",
              "sortable": false,
              "width": "50px",
            }
          }
          key="image"
        >
          <Avatar
            image="https://twitter-avatar.now.sh/AntoGriezmann"
            size="medium"
            type="user"
          />
        </styled.td>
        <styled.td
          column={
            Object {
              "field": "name",
              "sortable": true,
              "title": "Name",
              "width": "30%",
            }
          }
          key="name"
        >
          Antoine Griezmann
        </styled.td>
        <styled.td
          column={
            Object {
              "field": "country",
              "sortable": false,
              "title": "Country",
            }
          }
          key="country"
        >
          ðŸ‡«ðŸ‡·
        </styled.td>
        <styled.td
          column={
            Object {
              "field": "goals",
              "sortable": false,
              "title": "Goals",
            }
          }
          key="goals"
        >
          4
        </styled.td>
        <styled.td
          column={
            Object {
              "field": "assists",
              "sortable": false,
              "title": "Assists",
            }
          }
          key="assists"
        >
          2
        </styled.td>
      </styled.tr>
      <styled.tr
        data-cosmos-key="table.row"
        key="row-3"
        onClick={[Function]}
      >
        <styled.td
          column={
            Object {
              "children": [Function],
              "field": "image",
              "sortable": false,
              "width": "50px",
            }
          }
          key="image"
        >
          <Avatar
            image="https://twitter-avatar.now.sh/ivanperisic44"
            size="medium"
            type="user"
          />
        </styled.td>
        <styled.td
          column={
            Object {
              "field": "name",
              "sortable": true,
              "title": "Name",
              "width": "30%",
            }
          }
          key="name"
        >
          Ivan PeriÅ¡iÄ‡
        </styled.td>
        <styled.td
          column={
            Object {
              "field": "country",
              "sortable": false,
              "title": "Country",
            }
          }
          key="country"
        >
          ðŸ‡­ðŸ‡·
        </styled.td>
        <styled.td
          column={
            Object {
              "field": "goals",
              "sortable": false,
              "title": "Goals",
            }
          }
          key="goals"
        >
          3
        </styled.td>
        <styled.td
          column={
            Object {
              "field": "assists",
              "sortable": false,
              "title": "Assists",
            }
          }
          key="assists"
        >
          1
        </styled.td>
      </styled.tr>
    </styled.tbody>
  </styled.table>
  <Component
    rows={
      Array [
        <styled.tr
          data-cosmos-key="table.row"
          onClick={[Function]}
        >
          <styled.td
            column={
              Object {
                "children": [Function],
                "field": "image",
                "sortable": false,
                "width": "50px",
              }
            }
          >
            <Avatar
              image="https://twitter-avatar.now.sh/HKane"
              size="medium"
              type="user"
            />
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "name",
                "sortable": true,
                "title": "Name",
                "width": "30%",
              }
            }
          >
            Harry Kane
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "country",
                "sortable": false,
                "title": "Country",
              }
            }
          >
            ðŸ‡¬ðŸ‡§
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "goals",
                "sortable": false,
                "title": "Goals",
              }
            }
          >
            6
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "assists",
                "sortable": false,
                "title": "Assists",
              }
            }
          >
            0
          </styled.td>
        </styled.tr>,
        <styled.tr
          data-cosmos-key="table.row"
          onClick={[Function]}
        >
          <styled.td
            column={
              Object {
                "children": [Function],
                "field": "image",
                "sortable": false,
                "width": "50px",
              }
            }
          >
            <Avatar
              image="https://twitter-avatar.now.sh/Romelu_lukaku9"
              size="medium"
              type="user"
            />
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "name",
                "sortable": true,
                "title": "Name",
                "width": "30%",
              }
            }
          >
            Romelu Lukaku
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "country",
                "sortable": false,
                "title": "Country",
              }
            }
          >
            ðŸ‡§ðŸ‡ª
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "goals",
                "sortable": false,
                "title": "Goals",
              }
            }
          >
            4
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "assists",
                "sortable": false,
                "title": "Assists",
              }
            }
          >
            1
          </styled.td>
        </styled.tr>,
        <styled.tr
          data-cosmos-key="table.row"
          onClick={[Function]}
        >
          <styled.td
            column={
              Object {
                "children": [Function],
                "field": "image",
                "sortable": false,
                "width": "50px",
              }
            }
          >
            <Avatar
              image="https://twitter-avatar.now.sh/AntoGriezmann"
              size="medium"
              type="user"
            />
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "name",
                "sortable": true,
                "title": "Name",
                "width": "30%",
              }
            }
          >
            Antoine Griezmann
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "country",
                "sortable": false,
                "title": "Country",
              }
            }
          >
            ðŸ‡«ðŸ‡·
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "goals",
                "sortable": false,
                "title": "Goals",
              }
            }
          >
            4
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "assists",
                "sortable": false,
                "title": "Assists",
              }
            }
          >
            2
          </styled.td>
        </styled.tr>,
        <styled.tr
          data-cosmos-key="table.row"
          onClick={[Function]}
        >
          <styled.td
            column={
              Object {
                "children": [Function],
                "field": "image",
                "sortable": false,
                "width": "50px",
              }
            }
          >
            <Avatar
              image="https://twitter-avatar.now.sh/ivanperisic44"
              size="medium"
              type="user"
            />
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "name",
                "sortable": true,
                "title": "Name",
                "width": "30%",
              }
            }
          >
            Ivan PeriÅ¡iÄ‡
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "country",
                "sortable": false,
                "title": "Country",
              }
            }
          >
            ðŸ‡­ðŸ‡·
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "goals",
                "sortable": false,
                "title": "Goals",
              }
            }
          >
            3
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "assists",
                "sortable": false,
                "title": "Assists",
              }
            }
          >
            1
          </styled.td>
        </styled.tr>,
      ]
    }
  >
    There are no items to display
  </Component>
  <Component
    rows={
      Array [
        <styled.tr
          data-cosmos-key="table.row"
          onClick={[Function]}
        >
          <styled.td
            column={
              Object {
                "children": [Function],
                "field": "image",
                "sortable": false,
                "width": "50px",
              }
            }
          >
            <Avatar
              image="https://twitter-avatar.now.sh/HKane"
              size="medium"
              type="user"
            />
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "name",
                "sortable": true,
                "title": "Name",
                "width": "30%",
              }
            }
          >
            Harry Kane
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "country",
                "sortable": false,
                "title": "Country",
              }
            }
          >
            ðŸ‡¬ðŸ‡§
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "goals",
                "sortable": false,
                "title": "Goals",
              }
            }
          >
            6
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "assists",
                "sortable": false,
                "title": "Assists",
              }
            }
          >
            0
          </styled.td>
        </styled.tr>,
        <styled.tr
          data-cosmos-key="table.row"
          onClick={[Function]}
        >
          <styled.td
            column={
              Object {
                "children": [Function],
                "field": "image",
                "sortable": false,
                "width": "50px",
              }
            }
          >
            <Avatar
              image="https://twitter-avatar.now.sh/Romelu_lukaku9"
              size="medium"
              type="user"
            />
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "name",
                "sortable": true,
                "title": "Name",
                "width": "30%",
              }
            }
          >
            Romelu Lukaku
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "country",
                "sortable": false,
                "title": "Country",
              }
            }
          >
            ðŸ‡§ðŸ‡ª
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "goals",
                "sortable": false,
                "title": "Goals",
              }
            }
          >
            4
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "assists",
                "sortable": false,
                "title": "Assists",
              }
            }
          >
            1
          </styled.td>
        </styled.tr>,
        <styled.tr
          data-cosmos-key="table.row"
          onClick={[Function]}
        >
          <styled.td
            column={
              Object {
                "children": [Function],
                "field": "image",
                "sortable": false,
                "width": "50px",
              }
            }
          >
            <Avatar
              image="https://twitter-avatar.now.sh/AntoGriezmann"
              size="medium"
              type="user"
            />
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "name",
                "sortable": true,
                "title": "Name",
                "width": "30%",
              }
            }
          >
            Antoine Griezmann
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "country",
                "sortable": false,
                "title": "Country",
              }
            }
          >
            ðŸ‡«ðŸ‡·
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "goals",
                "sortable": false,
                "title": "Goals",
              }
            }
          >
            4
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "assists",
                "sortable": false,
                "title": "Assists",
              }
            }
          >
            2
          </styled.td>
        </styled.tr>,
        <styled.tr
          data-cosmos-key="table.row"
          onClick={[Function]}
        >
          <styled.td
            column={
              Object {
                "children": [Function],
                "field": "image",
                "sortable": false,
                "width": "50px",
              }
            }
          >
            <Avatar
              image="https://twitter-avatar.now.sh/ivanperisic44"
              size="medium"
              type="user"
            />
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "name",
                "sortable": true,
                "title": "Name",
                "width": "30%",
              }
            }
          >
            Ivan PeriÅ¡iÄ‡
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "country",
                "sortable": false,
                "title": "Country",
              }
            }
          >
            ðŸ‡­ðŸ‡·
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "goals",
                "sortable": false,
                "title": "Goals",
              }
            }
          >
            3
          </styled.td>
          <styled.td
            column={
              Object {
                "field": "assists",
                "sortable": false,
                "title": "Assists",
              }
            }
          >
            1
          </styled.td>
        </styled.tr>,
      ]
    }
  />
</styled.div>
`;
